apps/platform/src/app/(main)/(settings)/[workspace]/page.tsx

```diff
@@ -85,6 +99,28 @@ export default function WorkspaceSettingsPage(): JSX.Element {
     })
   )
 
+  const handleLeaveWorkspace = useCallback(async () => {
+    if (!selectedWorkspace) return
+    
+    setIsLeavingWorkspace(true)
+    const loadingToast = toast.loading('Leaving workspace...')
+    
+    try {
+      const { success } = await leaveWorkspace()
+      
+      if (success) {
+        setAllWorkspaces((prev) => prev.filter(w => w.id !== selectedWorkspace.id))
+        router.push('/')
+        toast.success('Successfully left the workspace')
+      }
+    } catch (error) {
+      toast.error('Failed to leave the workspace')
```

Comment:
You won't actually need to handle failure cases since that's already being handled in the useHttp hook